ansible/tasks/oauth2-proxy.yml

```diff
@@ -0,0 +1,29 @@
+- name: "download oauth2_proxy"
+  get_url:
+    src: "https://github.com/bitly/oauth2_proxy/releases/download/v2.2/oauth2_proxy-2.2.0.linux-amd64.go1.8.1.tar.gz"
+    dest: "/tmp/oauth2_proxy.tar.gz"
+    checksum: "sha256:1c16698ed0c85aa47aeb80e608f723835d9d1a8b98bd9ae36a514826b3acce56"
+
+- name: "unarchive oauth2_proxy"
+  unarchive:
```

Comment:
Can you verify whether this also works in check mode (because the `src` won't exist)?